python/paddle/fluid/dygraph/io.py

```diff
@@ -0,0 +1,773 @@
+        model_filename = None
+        params_filename = None
+        separate_params = False
+        if configs is not None:
+            model_filename = configs.model_filename
+            params_filename = configs.params_filename
+            separate_params = configs.separate_params
+
+        # 1. load program desc & construct _ProgramHolder
+        programs = _construct_program_holders(model_path, model_filename)
+
+        # 2. load layer parameters & parameter attirbutes
+        persis_vars = _construct_params_and_buffers(
+            model_path, programs, separate_params, params_filename)
+
+        # 3. construct TranslatedLayer object
+        translated_layer = TranslatedLayer(programs, persis_vars)
+
+        # 4. create TranslatedLayer's execution method
+        for method_name, program_holder in programs.items():
+            setattr(TranslatedLayer, method_name,
+                    TranslatedLayer._execution_method_creator(method_name,
+                                                              program_holder))
+
+        # 5. set TranslatedLayer's default mode to eval
+        translated_layer.eval()
+
+        return translated_layer
+
+    @staticmethod
+    def _execution_method_creator(method_name, program_holder):
+        def __impl__(self, *input):
+            # 1. prepare inputs, outputs, attrs
+            input_vars = []
+            for i, value in enumerate(input):
+                if not isinstance(value, (np.ndarray, core.VarBase)):
+                    raise TypeError(
+                        "The type of input in TranslatedLayer must be numpy array or Variable(VarBase), but received %s."
+                        % type(value))
+                # NOTE: In order to unify the API, firstly convert the input to VarBase
+                if isinstance(value, np.ndarray):
+                    var = core.VarBase(
+                        value=value,
+                        name=program_holder.input_names[i],
+                        persistable=False,
+                        place=framework._current_expected_place(),
+                        zero_copy=True)
+                else:
+                    var = value
+                    # NOTE: we changed var name here, 
+                    # but it may be an important name set by user
+                    var.name = program_holder.input_names[i]
+                input_vars.append(var)
+
+            persis_vars = []
+            for var_name in program_holder.persistable_names:
+                if var_name in self._parameters:
+                    persis_vars.append(self._parameters[var_name])
+                elif var_name in self._buffers:
+                    persis_vars.append(self._buffers[var_name])
+                else:
+                    raise ValueError(
+                        "The persistable variable %s is not exists in current TranslatedLayer."
+                        % var_name)
```

Comment:
If provided redundant parameters， we raise error directly here, or we can  warn user which parameters is not needed but allow user train it?

Reply:
under what circumstances will there be redundant parameters?